File: apps/HSMS-fe/src/app/models/Company.dto.ts

export interface CompanyMastersDto {
   id: number;
  companyname: string;
  registrationNumber: string;
  gstin: string;
  panNumber: string;
  email: string;
  phone: string;
  website?: string | null;
  addressLine1: string;
  addressLine2?: string | null;
  cityId: number;
  stateId: number;
  countryId: number;
  cityName: string;
  stateName: string;
  countryName: string;
  postalCode?: string | null;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;

}

export interface CreateCompanyDto {
  isActive: boolean;
  companyname: string;
  registrationNumber: string;
  gstin: string;
  panNumber: string;
  email: string;
  phone: string;
  website: string;
  addressLine1: string;
  addressLine2?: string;
  cityId: number;
  stateId: number;
  countryId: number;
  postalCode: string;
}

File: libs/contracts/Dto/CompanyDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HSMS.contracts.Dto
{
    public class CompanyDto
    {
        public int Id { get; set; }
        public string Companyname { get; set; } = string.Empty;
        public string RegistrationNumber { get; set; } = string.Empty;
        public string? Gstin { get; set; }
        public string? PanNumber { get; set; }
        public string Email { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public string? Website { get; set; }
        public string AddressLine1 { get; set; } = string.Empty;
        public string? AddressLine2 { get; set; }
        public int? CityId { get; set; }
        public int? StateId { get; set; }
        public int? CountryId { get; set; }
        public string? CityName { get; set; }
        public string? StateName { get; set; }
        public string? CountryName { get; set; }
        public string? PostalCode { get; set; }
        public bool IsActive { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
        public DateTimeOffset UpdatedAt { get; set; }
    }
}
